src/server/detail/snapshot_storage.h

```diff
@@ -42,10 +42,12 @@ class SnapshotStorage {
   virtual io::ReadonlyFileOrError OpenReadFile(const std::string& path) = 0;
 
   // Returns the path of the RDB file or DFS summary file to load.
-  virtual std::string LoadPath(const std::string_view& dir, const std::string_view& dbfilename) = 0;
+  virtual io::Result<std::string, GenericError> LoadPath(const std::string_view& dir,
+                                                         const std::string_view& dbfilename) = 0;
```

Comment:
Small side note, its okay to pass string_views by value 😅

Reply:
and please also use them instead of const string& below where its possible

Reply:
Sure, out of interest why is `string_view` prefered to `const std::string&`?

Reply:
string_view as an argument in function is usually better because it accepts both strings and const char* without doing any allocations. It's like a decorator around an immutable string passed to a function. When you use `const string` it accepts strings of course but if you pass `"foobar"` it will create a temporary string object during the call